Toric/ConvexAnalysis/PolyhedralCone.lean

```diff
@@ -19,11 +19,27 @@ namespace PointedCone
 section OrderedSemiring
 variable [Semiring R] [PartialOrder R] [IsOrderedRing R] [AddCommMonoid E] [Module R E] {s : Set E}
 
+theorem span_le (c : PointedCone R E) {s : Set E} :
+    span R s ≤ c ↔ s ⊆ c :=
+  Submodule.span_le
+
 /-- A pointed cone is polyhedral if it is the convex hull of finitely many points. -/
 def IsPolyhedral (c : PointedCone R E) : Prop := ∃ t : Finset E, PointedCone.span R t = c
 
 protected lemma IsPolyhedral.span (h : s.Finite) : (span R s).IsPolyhedral := ⟨h.toFinset, by simp⟩
 
+def isPolyhedral_iff_eq_span (c : PointedCone R E) :
+    c.IsPolyhedral ↔ ∃ t : Finset E, c = PointedCone.span R (t ∪ {0}) := by
+  apply Iff.intro
+  · rintro ⟨g, hg⟩
+    refine ⟨g, ?_⟩
+    apply le_antisymm
+    · simp [hg]
+    · simp only [Set.union_singleton, Submodule.span_insert_zero, hg, le_refl]
+  · rintro ⟨g, hg⟩
+    refine ⟨open Classical in g ∪ {0}, ?_⟩
+    simp_all
+
```

Comment:
I find it a bit strange that this is necessary. If it is, maybe it's better to inline this proof into `IsPolyhedral.isPolyhedral_span_of_isExtreme`?

Reply:
Inlining the direction I needed worked.